libtcmu_log.c

```diff
@@ -536,7 +609,8 @@ static int tcmu_log_dir_set(const char *log_dir)
 
 	tcmu_log_dir_free();
 	tcmu_log_dir = new_dir;
-	return 0;
+
+	return tcmu_logrotate_conf_set(tcmu_log_dir);
```

Comment:
I think you need to put this call before we free the old and set the new tcmu_log_dir in case the call to tcmu_logrotate_conf_set fails. We then do not have to worry about unraveling what was done here.